modules/dnn/src/layers/fast_convolution/fast_convolution.cpp

```diff
@@ -0,0 +1,693 @@
+    }
+
+#if CV_NEON
+    if ( conv->ifWinograd63
+         && inputShape[2] > 12 && inputShape[3] > 12
+         && inputShape[2] < 120 && inputShape[3] < 120 )
+    {
+        // In general, for winograd branch, more cores will give better performance.
+        int maxNumThread = std::max(getNumThreads(), 1);
+        if (runWinograd63(input, output, conv, maxNumThread, minval, maxval, activ, ifMinMaxAct))
+            return;
+    }
+#endif
+
+    float* inp = input.ptr<float>();
+    float* out = output.ptr<float>();
+
+    int N = inputShape[0], C = inputShape[1], Hi = inputShape[2], Wi = inputShape[3];  // [N, C, H, W]
+    int K = conv->K, Hk = conv->Hk, Wk = conv->Wk;
+    int H0 = outputShape[2], W0 = outputShape[3], ngroups = conv->ngroups;         // ngroups
+    int Cg = C/ngroups, Kg = K/ngroups;
+    int Kg_nblocks = (Kg + FAST_CONV_MR-1)/FAST_CONV_MR, Kg_aligned = Kg_nblocks*FAST_CONV_MR; // align to MR
+
+    const size_t inp_planesize = (size_t)Hi*Wi;
+    const size_t out_planesize = (size_t)H0*W0;
+
+    int pad_top = conv->pad_top, pad_bottom = conv->pad_bottom;
+    int pad_left = conv->pad_left;
+    int pad_right = conv->pad_right;
+
+    int stride_y = conv->stride_y, stride_x = conv->stride_x;
+    int dilation_y = conv->dilation_y, dilation_x = conv->dilation_x;
+
+    int ksize = Hk * Wk;
+    bool s1d1 = stride_x == 1 && stride_y == 1 && dilation_x == 1 && dilation_y == 1;
+    bool s1d1p0 = s1d1 && pad_top == 0 && pad_left ==0 && pad_bottom == 0 && pad_right == 0;
+    bool fast_1x1 = stride_x == 1 && stride_y == 1 && ksize == 1;
+    int HkWkCg = Hk*Wk*Cg;
+
+    enum { VEC_ALIGN = 8, DFT_TYPE = CV_32F };
+    size_t taskbufsize = FAST_CONV_NR*HkWkCg; // input buffer
+    size_t taskbufsizeOutput = FAST_CONV_NR * FAST_CONV_MR;
+    size_t inputbufsize = 0;
+    size_t outbufsize = ntasks * taskbufsizeOutput;
+
+    int stripes_per_sample = (out_planesize + FAST_CONV_NR - 1)/FAST_CONV_NR; // align to NR
+    size_t hw_task = stripes_per_sample;
+    size_t hw_aligned = stripes_per_sample * FAST_CONV_NR;
+
+    bool separatedLoop = false;
+
+    if (stripes_per_sample < 4 * ntasks)
+    {
+        // If stripes_per_sample is small, we parallelize on K (output channel).
+        stripes_per_sample = 1;
+
+        // Separated Parallelloop could save much time in packing input data. But it may cost more memory, we use it when batch size is 1.
+        if (N == 1)
+        {
+            separatedLoop = true;
+            inputbufsize = ngroups * hw_aligned * HkWkCg;
+        }
+
+        if (!separatedLoop)
+        {
+            inputbufsize = taskbufsize * ntasks;
+        }
+    }
+    else
+    {
+        // If stripes_per_sample is big, we parallelize on H0*W0.
+        Kg_nblocks = 1;
+        inputbufsize = taskbufsize * ntasks;
+    }
+
+    int Kstripes = Kg_nblocks*stripes_per_sample;
+    int nsubtasks = N*ngroups*Kstripes;
+
+    float* inpbuf_all = (float *)fastMalloc(inputbufsize * sizeof(float ));
```

Comment:
In Ficus engine I had to use C, because this is the native output language for Ficus compiler. In C++ code, please, **never** use plain "malloc" or its alternative. Use `std::vector<>` instead.